pkg/streamer/file.go

```diff
@@ -186,141 +187,90 @@ func fileSizeUpdated(path string, latestSize int64) (int, error) {
 	if err != nil {
 		return 0, terror.ErrGetRelayLogStat.Delegate(err, path)
 	}
-	currSize := fi.Size()
+	curSize := fi.Size()
 	switch {
-	case currSize == latestSize:
+	case curSize == latestSize:
 		return 0, nil
-	case currSize > latestSize:
-		log.L().Debug("size of relay log file has been changed", zap.String("file", path), zap.Int64("old size", latestSize), zap.Int64("size", currSize))
+	case curSize > latestSize:
+		log.L().Debug("size of relay log file has been changed", zap.String("file", path),
+			zap.Int64("old size", latestSize), zap.Int64("size", curSize))
 		return 1, nil
 	default:
-		log.L().Error("size of relay log file has been changed", zap.String("file", path), zap.Int64("old size", latestSize), zap.Int64("size", currSize))
+		log.L().Error("size of relay log file has been changed", zap.String("file", path),
+			zap.Int64("old size", latestSize), zap.Int64("size", curSize))
 		return -1, nil
 	}
 }
 
-// relaySubDirUpdated checks whether the relay sub directory updated
-// including file changed, created, removed, etc.
-func relaySubDirUpdated(ctx context.Context, watcherInterval time.Duration, dir string,
+// relayLogUpdatedOrNewCreated checks whether current relay log file is updated or new relay log is created.
+// we check the size of the file first, if the size is the same as the latest file, we assume there is no new write
+// so we need to check relay meta file to see if the new relay log is created.
+// this func will be blocked until current filesize changed or meta file updated or context cancelled.
+func relayLogUpdatedOrNewCreated(ctx context.Context, watcherInterval time.Duration, dir string,
 	latestFilePath, latestFile string, latestFileSize int64, updatePathCh chan string, errCh chan error) {
-	var err error
-	defer func() {
-		if err != nil {
-			errCh <- err
-		}
-	}()
-
-	// create polling watcher
-	watcher2 := watcher.NewWatcher()
-
-	// Add before Start
-	// no need to Remove, it will be closed and release when return
-	err = watcher2.Add(dir)
-	if err != nil {
-		err = terror.ErrAddWatchForRelayLogDir.Delegate(err, dir)
-		return
-	}
-
-	err = watcher2.Start(watcherInterval)
-	if err != nil {
-		err = terror.ErrWatcherStart.Delegate(err, dir)
-		return
-	}
-	defer watcher2.Close()
-
-	type watchResult struct {
-		updatePath string
-		err        error
-	}
-
-	result := make(chan watchResult, 1) // buffered chan to ensure not block the sender even return in the halfway
-	newCtx, cancel := context.WithCancel(ctx)
-	defer cancel()
-	go func() {
-		for {
-			select {
-			case <-newCtx.Done():
-				result <- watchResult{
-					updatePath: "",
-					err:        newCtx.Err(),
-				}
+	ticker := time.NewTicker(watcherInterval)
```

Comment:
This ticker is used only once? Maybe `time.After` is enough?

Reply:
when the binlog file size does not change at first and the relay meta does not move forward
will rely on the ticker to cycle check the binlogfile file size

Reply:
Sorry. I take this look mistakenly.